internal/protocol/protocol.go

```diff
@@ -59,7 +59,7 @@ type ConnectionID uint64
 type StreamID uint64
 
 // A ByteCount in QUIC
-type ByteCount uint64
+type ByteCount = uint64
```

Comment:
We introduced the ByteCount type so that we get compiler errors on invalid casts & conversions. This change defeats that purpose, so I'm against it. I don't think it's necessary either, we can just add the cast wherever CloseRemote is called.

Reply:
The problem is that if one (without this change) defines an interface with `CloseRemote(uint64)`, it does NOT match any defined function with `CloseRemote(ByteCount)`. E.g. if one tries to copy h2quic (e.g. to adjust it to a similar protocol) one has a problem, since one can neither import `protocol` nor can one define an interface which would match `CloseRemote(ByteCount)`.

Reply:
Reading you comment again, I think I understood it wrong. Did you mean to just change the function signature to `CloseRemote(uint64)`, cast the param and then keep using `ByteCount` everywhere else?

Reply:
Yes, change it to CloseRemote(uint64). CloseRemote is a layer violation by gQUIC (which is why it's not part of the public interface), it will go away with the new IETF HTTP mapping.